src/plugins/schema.js

```diff
@@ -0,0 +1,283 @@
+    return invalids.size ? invalids : null
+},
+  normalize: (transform, inline, invalids) => {
+    return invalids.reduce((t, n) => t.removeNodeByKey(n.key), transform)
+  }
+}
+
+/**
+ * A default schema rule to ensure that inline nodes are not empty
+ *
+ * @type {Object}
+ */
+
+const INLINE_NO_EMPTY = {
+  match: (object) => {
+    return object.kind == 'inline'
+  },
+  validate: (inline) => {
+    return inline.text == ''
+  },
+  normalize: (transform, node) => {
+    return transform.removeNodeByKey(node.key)
+  }
+}
+
+/**
+ * A default schema rule to ensure that void nodes contain a single space of content.
+ *
+ * @type {Object}
+ */
+
+const INLINE_VOID_TEXT_RULE = {
+  match: (object) => {
+    return (object.kind == 'inline' || object.kind == 'block') && object.isVoid
+  },
+  validate: (node) => {
+    return node.text !== ' ' || node.nodes.size !== 1
+  },
+  normalize: (transform, node) => {
+      transform = node.nodes.reduce((t, child) => {
+          return t.removeNodeByKey(child.key)
+      }, transform)
+
+      return transform.insertNodeByKey(node.key, 0, Text.createFromString(' '))
+  }
+}
+
+/**
+ * A default schema rule to ensure that inline void nodes are surrounded with text nodes
+ *
+ * @type {Object}
+ */
+
+const INLINE_VOID_TEXTS_AROUND_RULE = {
+  match: (object) => {
+    return object.kind == 'block'
+  },
+  validate: (block) => {
+    const invalids = block.nodes.reduce((accu, child, index) => {
+      if (child.kind === 'block' || !child.isVoid) {
+        return accu
+      }
+
+      const prevNode = index > 0 ? block.nodes.get(index - 1) : null
+      const nextNode = block.nodes.get(index + 1)
+
+      const prev = (!prevNode || isInlineVoid(prevNode))
+      const next = (!nextNode || isInlineVoid(nextNode.kind))
+
+      if (next || prev) {
+        accu.push({ next, prev, index })
+      }
+
+      return accu
+    }, [])
+
+    return invalids.length ? invalids : null
+  },
+  normalize: (transform, block, invalids) => {
+    return invalids.reduce((t, { index, next, prev }) => {
+      if (prev) t = transform.insertNodeByKey(block.key, index, Text.create())
+      if (next) t = transform.insertNodeByKey(block.key, index + 1, Text.create())
+      return t
+    }, transform)
+  }
+}
+
+/**
+ * Join adjacent text nodes.
+ *
+ * @type {Object}
+ */
+
+const NO_ADJACENT_TEXT_RULE = {
+  match: (object) => {
+    return object.kind == 'block' || object.kind == 'inline'
+  },
+  validate: (node) => {
+    const { nodes } = node
+    const invalids = nodes
+      .map((n, i) => {
+        const next = nodes.get(i + 1)
+        if (n.kind !== 'text' || !next || next.kind !== 'text') {
+          return
+        }
+
+        return [n, next]
+      })
+      .filter(Boolean)
+
+    return invalids.size ? invalids : null
+  },
+  normalize: (transform, node, pairs) => {
+    return pairs
+        // We reverse the list since we want to handle 3 consecutive text nodes
```

Comment:
Because `joinNodeByKey` merges the second node into the first node, if I understand well.